pkg/hubble/dropeventemitter/dropeventemitter.go

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package dropeventemitter
+
+import (
+	"context"
+	"strconv"
+
+	v1 "k8s.io/api/core/v1"
+	typedv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/record"
+
+	slimv1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1"
+	metaslimv1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/apis/meta/v1"
+	slimscheme "github.com/cilium/cilium/pkg/k8s/slim/k8s/client/clientset/versioned/scheme"
+
+	flowpb "github.com/cilium/cilium/api/v1/flow"
+	"github.com/cilium/cilium/pkg/identity"
+	client "github.com/cilium/cilium/pkg/k8s/client"
+	"github.com/cilium/cilium/pkg/time"
+)
+
+type DropEventEmitter struct {
+	recorder record.EventRecorder
+}
+
+func NewDropEventEmitter(interval time.Duration, k8s client.Clientset) *DropEventEmitter {
+	broadcaster := record.NewBroadcasterWithCorrelatorOptions(record.CorrelatorOptions{
+		QPS:                  1 / float32(interval.Seconds()),
+		MaxEvents:            1,
+		MaxIntervalInSeconds: int(interval.Seconds()),
+	})
+	broadcaster.StartRecordingToSink(&typedv1.EventSinkImpl{Interface: k8s.CoreV1().Events("")})
+
+	return &DropEventEmitter{
+		recorder: broadcaster.NewRecorder(slimscheme.Scheme, v1.EventSource{Component: "cilium"}),
+	}
+}
+
+func (e *DropEventEmitter) ProcessFlow(ctx context.Context, flow *flowpb.Flow) error {
+	// Only handle packet drops due to policy related to a Pod
+	if flow.Verdict != flowpb.Verdict_DROPPED ||
+		flow.DropReasonDesc != flowpb.DropReason_POLICY_DENIED ||
+		(flow.TrafficDirection == flowpb.TrafficDirection_INGRESS &&
+			flow.Destination.PodName == "") ||
+		(flow.TrafficDirection == flowpb.TrafficDirection_EGRESS &&
+			flow.Source.PodName == "") {
+		return nil
+	}
```

Comment:
The helm doc for the feature states "Emit packet drop Events related to pods" but the codes seems to filter for specific types of drops.

What about, instead of hardcoding specific filters here, we make it configurable by the user with some reasonable default?
I'm thinking of something like that
```yaml
  # -- Emit v1.Events related to pods on detection of packet drops.
  dropEventEmitter::
    enabled: false
    # --- Minimum time between emitting same events.
    interval: 120s
    # --- Reason for the flow being dropped
    dropReason:
      - auth_required
      - policy_denied
      - ...
```

Reply:
Interesting suggestion, I'll look into that.

Reply:
Option added, thanks.